sklearn/cluster/dbscan_.py

```diff
@@ -194,6 +207,10 @@ class DBSCAN(BaseEstimator, ClusterMixin):
     Notes
     -----
     See examples/plot_dbscan.py for an example.
+    
+    This implementation bulk-computes all neighborhood queries. This yields a
+    worst case memory complexity of O(n^2), compared to the O(n) memory complexity
```

Comment:
Perhaps we should just state it as `O(n.d)` compared to `O(d)` for d being the average density; and then we can say "average case" or "in all cases".